### PR TITLE
React to changes kvm.sh for nightly an other versions

### PR DESCRIPTION
**Pull request by vlesierse**
- Use KVM_USER_HOME instead of KRE_USER_HOME
- Change folder from packages to runtimes

### CODE REVIEW

**Comment by vlesierse** [comment]
Looks like beta 4 breaks the test application..... Good news is that the image builds again.

**Comment by ahmetalpbalkan** [comment]
@vlesierse if you wanna debug, you can see the failed container's logs here https://circle-artifacts.com/gh/aspnet/aspnet-docker/5/artifacts/0/home/ubuntu/aspnet-docker/container-logs/345773ce8cdb.log

**Comment by vlesierse** [comment]
Thnx. I need to adjust the ci script to checkout the dev branch of the home repo before building the image. This branch has the proper samples to work with the latest dev version.

File: .ci/run-with-all-tags.sh [hunk 1]
```diff
@@ -7,9 +7,11 @@ set -e 	# Exit immediately upon failure
 
 for tag in `.ci/find-tags.sh $1`; do
 	TAG=${IMAGE}:${tag}
+	BRANCH=master
+	if [[ ${tag} == nightly ]]; then BRANCH=dev; fi
```
Comment (ahmetalpbalkan) [inline .ci/run-with-all-tags.sh]: `if [[ "$tag" == "nightly" ]]` would be better here.

File: nightly/Dockerfile [hunk 2]
```diff
@@ -6,9 +6,9 @@ ENV KRE_USER_HOME /opt/kre
 RUN apt-get -qq update && apt-get -qqy install unzip
 
 ONBUILD RUN curl -sSL https://raw.githubusercontent.com/aspnet/Home/dev/kvminstall.sh | sh
-ONBUILD RUN bash -c "source $KRE_USER_HOME/kvm/kvm.sh \
+ONBUILD RUN bash -c "KVM_USER_HOME=$KRE_USER_HOME && source $KRE_USER_HOME/kvm/kvm.sh \
 	&& kvm install latest -a default \
-	&& kvm alias default | xargs -i ln -s $KRE_USER_HOME/packages/{} $KRE_USER_HOME/packages/default"
+	&& kvm alias default | xargs -i ln -s $KRE_USER_HOME/runtimes/{} $KRE_USER_HOME/runtimes/default"
```
Comment (ahmetalpbalkan) [inline nightly/Dockerfile]: is the packages→runtimes change something comes with newer versions of kvm? I haven't really looked into these.

File: .ci/build-app-image.sh [hunk 3]
```diff
@@ -4,13 +4,17 @@ set -e 	# Exit immediately upon failure
...
 
 BASE_IMAGE=$1
 TEST_APP=$2
 TAG=$3
+BRANCH=$4
 
 echo "[CI] Injecting Dockerfile to project $TEST_APP..."
 cd  $SAMPLES_REPO/samples/$TEST_APP
+echo "[CI] Checkout branch $BRANCH..."
+git checkout $BRANCH
```
Comment (ahmetalpbalkan) [inline .ci/build-app-image.sh]: could you also add  `git checkout -` after this command finishes? That could make sure we switch back to the old branch if we are done with this script successfully.

**Comment by vlesierse** [comment]
I agree with the code commends and will change them. Tonight I've tried to get the samples running within the Docker container, but I keep running into this error. Really annoying.

**Comment by ahmetalpbalkan** [comment]
@vlesierse [this](https://circle-artifacts.com/gh/aspnet/aspnet-docker/9/artifacts/0/home/ubuntu/aspnet-docker/container-logs/e01f88e03373.log) looks like something asp.net folks might know. try opening an issue or just head to jabbr.net aspnetvnext channel to ask what's going on.

**Comment by vlesierse** [comment]
Thnx, I've added an [issue](https://github.com/aspnet/Home/issues/332)

**Comment by vlesierse** [comment]
It appeared that the update to beta 3 broke the other versions as well. I had to make the same adjustments to get this to pass the tests.

There are a lot of commits now. I will squash them when it's ok and ready to merge.

File: nightly/Dockerfile [hunk 4]
```diff
@@ -23,4 +23,6 @@ RUN LIBUV_VERSION=1.0.0-rc2 \
 	&& rm -rf /usr/local/src/libuv-$LIBUV_VERSION \
 	&& ldconfig
 
-ENV PATH $PATH:$KRE_USER_HOME/packages/default/bin
+COPY NuGet.Config /root/.config/NuGet/
```
Comment (ahmetalpbalkan) [inline nightly/Dockerfile]: could this be $HOME instead of /root/? We stopped using /root/ a while ago.

**Comment by ahmetalpbalkan** [comment]
@vlesierse it seems like CI is green even on nightly now! that's wonderful. wanna squash your commits into one and git push -f to your branch to update the pr now?

**Comment by vlesierse** [comment]
Done

**Comment by ahmetalpbalkan** [comment]
@vlesierse thanks. do beta1/beta2 images also need  to be updated with `curl ... | KRE_USER_HOME=$KVM_USER_HOME .... sh` ? or is it just the dev branch? 

normally, since we set the env variable at the beginning, `anything | sh` should carry the environment variable over to `sh`, I'm not sure why this is necessary. Try this:

```sh
$ export A=B
$ echo "env | grep ^A=" | sh
A=B
```

**Comment by vlesierse** [comment]
No, only for the nightly and maybe for beta 3. The older beta releases downloaded the wrong kvm.sh from the release branch and broke the build. That one is also fixed.

I will do the change tomorrow so that it can be merged.

**Comment by vlesierse** [comment]
The actual reason of the `KRE_USER_HOME=$KVM_USER_HOME` is because the `kvminstall.sh` requires the `KRE_USER_HOME` variable to be set and `kvm.sh` requires `KVM_USER_HOME` to be set (at least for the nightly and not beta1 or beta2).

I've tried putting the environment variables in front of the ` | ` but it doesn't take the variable when executing the script.

This line is actually taken from the [ASP.NET 5 Home](https://github.com/aspnet/Home#linux) page.

I expect another naming change to `XRE` and dotnetsdk (once final).

**Comment by ahmetalpbalkan** [comment]
@vlesierse oh sorry that is my bad. I read KRE_USER_HOME and KVM_USER_HOME as the same things. From beta3 and onwards we must start that environment variable and call it match KRE_USER_HOME.

Yes there'll be probably some huge naming changes that's going to break us, I'm excited about that, too. :joy:

LGTM. Thanks!

**Comment by vlesierse** [comment]
@ahmetalpbalkan I see the CI build on the master branch is failing, but it looks like something is going wrong internally in ASP.NET/.NET.

**Comment by ahmetalpbalkan** [comment]
Yeah that's certainly interesting since the same build was passing on PR and I have rebuilt failing build several times now. I can see it's mono internal error but wasn't happening on PR branch. Clearly interesting.

**Comment by ahmetalpbalkan** [comment]
Here's the failure while starting the app, I'm starting to rebuild your PR CI as well to see if it's a merge-related issue or a dependency changed meanwhile:

```
2015-02-25T16:45:58.971901332Z Information: [NuGetAssemblyLoader]: Loaded name=Microsoft.AspNet.Cryptography.Internal in 0ms
2015-02-25T16:45:58.972138663Z Can't find custom attr constructor image: /root/.k/packages/Microsoft.AspNet.Cryptography.Internal/1.0.0-beta4-10918/lib/aspnet50/Microsoft.AspNet.Cryptography.Internal.dll mtoken: 0x0a000004
2015-02-25T16:45:58.972138663Z * Assertion at class.c:5695, condition `!mono_loader_get_last_error ()' not met
2015-02-25T16:45:58.972138663Z 
2015-02-25T16:45:58.972209193Z Stacktrace:
2015-02-25T16:45:58.972209193Z 
2015-02-25T16:45:58.972209193Z   at <unknown> <0xffffffff>
2015-02-25T16:45:58.972209193Z   at Microsoft.Framework.DependencyInjection.DataProtectionServiceCollectionExtensions.AddDataProtection (Microsoft.Framework.DependencyInjection.IServiceCollection,Microsoft.Framework.ConfigurationModel.IConfiguration) <0x00063>
2015-02-25T16:45:58.972264013Z   at Microsoft.Framework.DependencyInjection.MvcServiceCollectionExtensions.ConfigureDefaultServices (Microsoft.Framework.DependencyInjection.IServiceCollection,Microsoft.Framework.ConfigurationModel.IConfiguration) <0x00032>
2015-02-25T16:45:58.972264013Z   at Microsoft.Framework.DependencyInjection.MvcServiceCollectionExtensions.AddMvc (Microsoft.Framework.DependencyInjection.IServiceCollection,Microsoft.Framework.ConfigurationModel.IConfiguration) <0x0001f>
2015-02-25T16:45:58.972322723Z   at Microsoft.Framework.DependencyInjection.MvcServiceCollectionExtensions.AddMvc (Microsoft.Framework.DependencyInjection.IServiceCollection) <0x00016>
2015-02-25T16:45:58.972322723Z   at HelloMvc.Startup/<>c.<Configure>b__0_0 (Microsoft.Framework.DependencyInjection.IServiceCollection) <0x00018>
2015-02-25T16:45:58.972322723Z   at Microsoft.AspNet.Builder.ContainerExtensions/<>c__DisplayClass3_0.<UseServices>b__0 (Microsoft.Framework.DependencyInjection.IServiceCollection) <0x00024>
2015-02-25T16:45:58.972374623Z   at Microsoft.AspNet.Builder.ContainerExtensions.UseServices (Microsoft.AspNet.Builder.IApplicationBuilder,System.Func`2<Microsoft.Framework.DependencyInjection.IServiceCollection, System.IServiceProvider>) <0x0003a>
2015-02-25T16:45:58.972374623Z   at Microsoft.AspNet.Builder.ContainerExtensions.UseServices (Microsoft.AspNet.Builder.IApplicationBuilder,System.Action`1<Microsoft.Framework.DependencyInjection.IServiceCollection>) <0x000fd>
2015-02-25T16:45:58.972405455Z   at HelloMvc.Startup.Configure (Microsoft.AspNet.Builder.IApplicationBuilder) <0x000ea>
2015-02-25T16:45:58.972422616Z   at (wrapper runtime-invoke) <Module>.runtime_invoke_void__this___object (object,intptr,intptr,intptr) <0xffffffff>
2015-02-25T16:45:58.972439059Z   at <unknown> <0xffffffff>
2015-02-25T16:45:58.972454797Z   at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&) <0xffffffff>
2015-02-25T16:45:58.972484615Z   at System.Reflection.MonoMethod.Invoke (object,System.Reflection.BindingFlags,System.Reflection.Binder,object[],System.Globalization.CultureInfo) <0x000f2>
2015-02-25T16:45:58.972507862Z   at System.Reflection.MethodBase.Invoke (object,object[]) <0x0002a>
2015-02-25T16:45:58.972527228Z   at Microsoft.AspNet.Hosting.Startup.StartupLoader.Invoke (System.Reflection.MethodInfo,object,Microsoft.AspNet.Builder.IApplicationBuilder,Microsoft.Framework.DependencyInjection.IServiceCollection) <0x0025b>
2015-02-25T16:45:58.972552958Z   at Microsoft.AspNet.Hosting.Startup.StartupLoader/<>c__DisplayClass5_0.<LoadStartup>b__2 (Microsoft.AspNet.Builder.IApplicationBuilder) <0x00203>
2015-02-25T16:45:58.972584029Z   at Microsoft.AspNet.Hosting.HostingEngine.EnsureApplicationDelegate (Microsoft.AspNet.Hosting.HostingContext) <0x00050>
2015-02-25T16:45:58.972605004Z   at Microsoft.AspNet.Hosting.HostingEngine.Start (Microsoft.AspNet.Hosting.HostingContext) <0x00085>
2015-02-25T16:45:58.972623971Z   at Microsoft.AspNet.Hosting.Program.Main (string[]) <0x004b2>
2015-02-25T16:45:58.972656786Z   at (wrapper runtime-invoke) <Module>.runtime_invoke_void__this___object (object,intptr,intptr,intptr) <0xffffffff>
2015-02-25T16:45:58.972675171Z   at <unknown> <0xffffffff>
2015-02-25T16:45:58.972690875Z   at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&) <0xffffffff>
2015-02-25T16:45:58.972716055Z   at System.Reflection.MonoMethod.Invoke (object,System.Reflection.BindingFlags,System.Reflection.Binder,object[],System.Globalization.CultureInfo) <0x000f2>
2015-02-25T16:45:58.972732926Z   at System.Reflection.MethodBase.Invoke (object,object[]) <0x0002a>
2015-02-25T16:45:58.972748153Z   at Microsoft.Framework.Runtime.Common.EntryPointExecutor.Execute (System.Reflection.Assembly,string[],System.IServiceProvider) <0x0012b>
2015-02-25T16:45:58.972766254Z   at Microsoft.Framework.ApplicationHost.Program.ExecuteMain (Microsoft.Framework.Runtime.DefaultHost,string,string[]) <0x0060a>
2015-02-25T16:45:58.972784786Z   at Microsoft.Framework.ApplicationHost.Program.Main (string[]) <0x004aa>
2015-02-25T16:45:58.972818403Z   at (wrapper runtime-invoke) <Module>.runtime_invoke_object__this___object (object,intptr,intptr,intptr) <0xffffffff>
2015-02-25T16:45:58.972840002Z   at <unknown> <0xffffffff>
2015-02-25T16:45:58.972855483Z   at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&) <0xffffffff>
2015-02-25T16:45:58.972872171Z   at System.Reflection.MonoMethod.Invoke (object,System.Reflection.BindingFlags,System.Reflection.Binder,object[],System.Globalization.CultureInfo) <0x000f2>
2015-02-25T16:45:58.972888558Z   at System.Reflection.MethodBase.Invoke (object,object[]) <0x0002a>
2015-02-25T16:45:58.972906921Z   at Microsoft.Framework.Runtime.Common.EntryPointExecutor.Execute (System.Reflection.Assembly,string[],System.IServiceProvider) <0x0012b>
2015-02-25T16:45:58.972926125Z   at kre.host.Bootstrapper.Main (string[]) <0x00364>
2015-02-25T16:45:58.972947523Z   at (wrapper runtime-invoke) <Module>.runtime_invoke_object__this___object (object,intptr,intptr,intptr) <0xffffffff>
2015-02-25T16:45:58.972979361Z   at <unknown> <0xffffffff>
2015-02-25T16:45:58.972979361Z   at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&) <0xffffffff>
2015-02-25T16:45:58.973008966Z   at System.Reflection.MonoMethod.Invoke (object,System.Reflection.BindingFlags,System.Reflection.Binder,object[],System.Globalization.CultureInfo) <0x000f2>
2015-02-25T16:45:58.973025074Z   at System.Reflection.MethodBase.Invoke (object,object[]) <0x0002a>
2015-02-25T16:45:58.973040658Z   at kre.hosting.RuntimeBootstrapper.ExecuteAsync (string[]) <0x01597>
2015-02-25T16:45:58.973059368Z   at kre.hosting.RuntimeBootstrapper.Execute (string[]) <0x00066>
2015-02-25T16:45:58.973075328Z   at EntryPoint.Main (string[]) <0x00320>
2015-02-25T16:45:58.973094177Z   at (wrapper runtime-invoke) <Module>.runtime_invoke_int_object (object,intptr,intptr,intptr) <0xffffffff>
2015-02-25T16:45:58.973110659Z 
2015-02-25T16:45:58.973110659Z Native stacktrace:
2015-02-25T16:45:58.973142328Z 
2015-02-25T16:45:58.974340437Z 	mono() [0x4accac]
2015-02-25T16:45:58.974369535Z 	/lib/x86_64-linux-gnu/libpthread.so.0(+0xf0a0) [0x7f2dcd9b80a0]
2015-02-25T16:45:58.974369535Z 	/lib/x86_64-linux-gnu/libc.so.6(gsignal+0x35) [0x7f2dcd650165]
2015-02-25T16:45:58.974399807Z 	/lib/x86_64-linux-gnu/libc.so.6(abort+0x180) [0x7f2dcd6533e0]
2015-02-25T16:45:58.974399807Z 	mono() [0x6232f9]
2015-02-25T16:45:58.974399807Z 	mono() [0x623507]
2015-02-25T16:45:58.974449822Z 	mono() [0x623656]
2015-02-25T16:45:58.974449822Z 	mono() [0x5182d7]
2015-02-25T16:45:58.974449822Z 	mono() [0x51808d]
2015-02-25T16:45:58.974490856Z 	mono(mono_field_from_token+0x153) [0x53e243]
2015-02-25T16:45:58.974490856Z 	mono() [0x456d19]
2015-02-25T16:45:58.974490856Z 	mono() [0x42c761]
2015-02-25T16:45:58.974531916Z 	mono() [0x42e2bd]
2015-02-25T16:45:58.974531916Z 	mono() [0x42ec4b]
2015-02-25T16:45:58.974531916Z 	mono() [0x4af15e]
2015-02-25T16:45:58.974569531Z 	[0x40a06166]
2015-02-25T16:45:58.976812805Z 
2015-02-25T16:45:58.976812805Z Debug info from gdb:
2015-02-25T16:45:58.976812805Z 
2015-02-25T16:45:58.976868833Z 
2015-02-25T16:45:58.976868833Z =================================================================
2015-02-25T16:45:58.976868833Z Got a SIGABRT while executing native code. This usually indicates
2015-02-25T16:45:58.976868833Z a fatal error in the mono runtime or one of the native libraries 
2015-02-25T16:45:58.976918862Z used by your application.
2015-02-25T16:45:58.976918862Z =================================================================
2015-02-25T16:45:58.976918862Z 
2015-02-25T16:45:59.102818497Z Aborted (core dumped)
```

**Comment by vlesierse** [comment]
Seems like the package has been changed over night when the build suddenly fails. The HelloWeb app does succeed.

**Comment by ahmetalpbalkan** [comment]
@vlesierse any ideas how we can locate this?
@davidfowl some change to `Microsoft.AspNet.Cryptography.Internal` last night suddenly started to throw SIGABRT on Mono consistently.

**Comment by vlesierse** [comment]
The changesets aren't that spectaculair.
https://github.com/aspnet/DataProtection/commit/ab18f52e987d301a5ba433cd025e690e00b5cdf9I
https://github.com/aspnet/DataProtection/commit/544c83812c1697db5257fbe40ff2598ac9cf923c

**Comment by ahmetalpbalkan** [comment]
@vlesierse let's take `nightly` from CI build. it doesn't make sense to have it there actually. entire point of nightly is to break somewhere :)

**Comment by vlesierse** [comment]
Yes, I totally agree on that. Nightlies intend to break.....

**Pull request merged**